genrules.cpp

```diff
@@ -3969,35 +3969,58 @@ int Game::GenRules(const AString &rules, const AString &css, const AString &intr
 	  << "EXCHANGE 3453 10 SWOR 10 LBOW\n"
 	  << example_end();
 
-	if (Globals->FACTION_LIMIT_TYPE == GameDefs::FACLIM_FACTION_TYPES) {
+	if (Globals->FACTION_LIMIT_TYPE == GameDefs::FACLIM_FACTION_TYPES)
+	{
 		f << enclose(class_tag("div", "rule"), true) << '\n' << enclose("div", false);
 		f << anchor("faction") << '\n';
+
+		const bool martial_faction = Globals->FACTION_ACTIVITY == FactionActivityRules::MARTIAL_MERGED;
```

Comment:
It looks like this test is slightly wrong as it's resulting in the neworigins code generating rules which reference war/trade/magic rather than martial/magic.

Reply:
you need to add parens around the (Globals->FACTION_ACTIIVITY == )..

Reply:
Ugh.. I'm an idiot.. the code is correct.. the snapshot test is diffing backwards which led me to read it wrong.. I will fix that in one of my upcoming commits.. Regardless you need to regenerate the rules snapshots and include them as part of the diff in order to allow the tests to pass.